packages/material-ui/src/Tooltip/useKeyboardFocus.js

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+
+let prepared = false;
+let keyboardModality = false;
+
+function handleKeyDown() {
+  keyboardModality = true;
+}
+
+function handlePointerDown() {
+  keyboardModality = false;
+}
+
+export function prepare() {
+  prepared = true;
+  document.addEventListener('keydown', handleKeyDown, { passive: true });
+  document.addEventListener('mousedown', handlePointerDown, { passive: true });
+  document.addEventListener('pointerdown', handlePointerDown, { passive: true });
+  document.addEventListener('touchstart', handlePointerDown, { passive: true });
+}
+
+export function teardown() {
+  document.removeEventListener('keydown', handleKeyDown, { passive: true });
+  document.removeEventListener('mousedown', handlePointerDown, { passive: true });
+  document.removeEventListener('pointerdown', handlePointerDown, { passive: true });
+  document.removeEventListener('touchstart', handlePointerDown, { passive: true });
+  prepared = false;
+}
+
+/**
+ * @returns true if called during a focus event that applied focus-visible
+ */
+export default function useKeyboardFocus() {
```

Comment:
What do you think of renaming it `useFocusVisible()`?

Reply:
It's not on feature parity with focus-visible. It would indicate we could just use this instead of ButtonBase/focusVisible.

Reply:
What behavior is missing to be feature parity?

Reply:
Considering the polyfill has 4 times more code probably a lot.

Reply:
The polyfill can't relies on :focus-visible as you do. It probably helps a lot.

Reply:
:focus-visible is only implemented in chrome and behind a flag. Most non-dev machines won't even use this fallback that I didn't even properly implement. This is a far cry from a proper focus-visible polyfill and I'm not sure why you pushing for it to be one?